src/lsp.sml

```diff
@@ -258,16 +258,23 @@ fun elabFile (state: state) (fileName: string): ({ decls: Elab.decl list, envBef
             end
     end
 
+fun uniq (comp: 'b -> 'b -> bool) (bs: 'b list) =
+    case bs of
+        [] => []
+      | (l as b :: bs') => b :: uniq comp (List.filter (comp b) bs')
```

Comment:
Did you mean to negate the filter condition here?  This function seems to be ensuring that all elements of the output list are equivalent (according to the provided relation).  Maybe you have only been testing it on lists of length at most one?

Reply:
Ok that was wrong... That'll teach me to blindly copy and paste from stack overflow.... And that also explains why I've been seeing only one error lately! :D